includes/Classifai/Providers/Azure/TextToSpeech.php

```diff
@@ -0,0 +1,673 @@
+		global $post;
+
+		if ( ! $post ) {
+			return $content;
+		}
+
+		if ( ! in_array( $post->post_type, self::get_supported_post_types(), true ) ) {
+			return $content;
+		}
+
+		/**
+		 * Filter to disable Text to Speech synthesis for a post by post ID.
+		 *
+		 * @param boolean 'is_disabled' Boolean to toggle the service. By default - false.
+		 * @param boolean 'post_id'     Post ID.
+		 */
+		if ( apply_filters( 'classifai_disable_post_to_audio_block', false, $post->ID ) ) {
+			return $content;
+		}
+
+		$audio_attachment_id = (int) get_post_meta( $post->ID, self::AUDIO_ID_KEY, true );
+
+		if ( ! $audio_attachment_id ) {
+			return $content;
+		}
+
+		wp_enqueue_script(
+			'classifai-post-audio-player-js',
+			CLASSIFAI_PLUGIN_URL . '/dist/post-audio-controls.js',
+			get_asset_info( 'post-audio-controls', 'dependencies' ),
+			get_asset_info( 'post-audio-controls', 'version' ),
+			true
+		);
+
+		wp_enqueue_style(
+			'classifai-post-audio-player-css',
+			CLASSIFAI_PLUGIN_URL . '/dist/post-audio-controls.css',
+			array(),
+			get_asset_info( 'post-audio-controls', 'version' ),
+			'all'
+		);
+
+		$audio_timestamp      = (int) get_post_meta( $post->ID, self::AUDIO_TIMESTAMP_KEY, true );
+		$audio_attachment_url = sprintf(
+			'%1$s?ver=%2$s',
+			wp_get_attachment_url( $audio_attachment_id ),
+			filter_var( $audio_timestamp, FILTER_SANITIZE_NUMBER_INT )
+		);
+
+		ob_start();
+
+		?>
+			<div>
+				<div class='classifai-listen-to-post-wrapper'>
+					<div class="class-post-audio-controls">
+						<span class="dashicons dashicons-controls-play"></span>
+						<span class="dashicons dashicons-controls-pause"></span>
+					</div>
+					<div class='classifai-post-audio-heading'>
+						<?php
+							$list_to_post_text = apply_filters( 'classifai_listen_to_this_post_text', esc_html__( 'Listen to this post', 'classifai' ) );
+							echo esc_html( $list_to_post_text );
+						?>
+					</div>
+				</div>
+				<audio id="classifai-post-audio-player" src="<?php echo esc_url( $audio_attachment_url ); ?>"></audio>
+			</div>
+		<?php
+
+		return ob_get_clean() . $content;
+	}
+
+	/**
+	 * Render the post types checkbox array.
+	 *
+	 * @param array $args Settings for the input
+	 *
+	 * @return void
+	 */
+	public function render_post_types_checkboxes( $args ) {
+		echo '<ul>';
+		$post_types = get_post_types_for_language_settings();
+
+		foreach ( $post_types as $post_type ) {
+			$args = [
+				'label_for'    => $post_type->name,
+				'option_index' => 'post_types',
+				'input_type'   => 'checkbox',
+			];
+
+			echo '<li>';
+			$this->render_input( $args );
+			echo '<label for="classifai-settings-' . esc_attr( $post_type->name ) . '">' . esc_html( $post_type->label ) . '</label>';
+			echo '</li>';
+		}
+
+		echo '</ul>';
+	}
+
+	/**
+	 * Generic text input field callback
+	 *
+	 * @param array $args The args passed to add_settings_field.
+	 */
+	public function render_input( $args ) {
```

Comment:
So we have this method in the parent class and seems to be pretty similar to what we have here. If there's specific changes made here that are needed, I'd suggest trying to make the parent method flexible enough to account for those instead of duplicating this method here